persistence/sql/persister_consent.go

```diff
@@ -438,3 +438,92 @@ func (p *Persister) VerifyAndInvalidateLogoutRequest(ctx context.Context, verifi
 		return nil
 	})
 }
+
+func (p *Persister) FlushInactiveLoginConsentRequests(ctx context.Context, notAfter time.Time) error {
+	/* #nosec G201 table is static */
+	var lr consent.LoginRequest
+	var lrh consent.HandledLoginRequest
+
+	var cr consent.ConsentRequest
+	var crh consent.HandledConsentRequest
+
+	// Delete all entries (and their FK)
+	// where hydra_oauth2_authentication_request were timed-out or rejected
+	// - hydra_oauth2_authentication_request_handled
+	// - hydra_oauth2_consent_request_handled
+	// AND
+	// - hydra_oauth2_authentication_request.requested_at < ttl.login_consent_request
+	// - hydra_oauth2_authentication_request.requested_at < notAfter
+
+	// Using NOT EXISTS instead of LEFT JOIN or NOT IN due to
+	// LEFT JOIN not supported by Postgres and NOT IN will have performance hits with large tables.
+	// https://stackoverflow.com/questions/19363481/select-rows-which-are-not-present-in-other-table/19364694#19364694
+	// Cannot use table aliasing in MYSQL, will work in Postgresql though...
+	err := p.Connection(ctx).RawQuery(fmt.Sprintf(`
```

Comment:
Both queries should be done in a transaction so that we roll back if an error occurrs and don't end up with a broken state!

Reply:
So wrap the calls in `p.transaction(ctx, func(ctx context.Context, c *pop.Connection) error {}` will execute it in the transaction?

Reply:
Yes, and it will also continue a transaction that is already running and stored in the context. So you can safely call other functions within the persister as well.